samples/account-linking/csharp/README.md

```diff
@@ -21,9 +23,103 @@ The code generically handles account linking for OAuth2.0, the only GitHub speci
+
+#### SSO Overview
+Teams enacts AAD SSO in the following manner:
+1.  In the tab, call getAuthToken() using the Teams JavaScript SDK. This tells Teams to obtain an auth token.
+2.  If this is the first time the current user has used your tab application, the user will be prompted to consent.
+3.  Microsoft Teams requests a token for the tab application from the Azure AD endpoint for the current user.
+4.  Azure AD sends the tab application token back to Teams.
+5.  Microsoft Teams then sends the token back to the tab application.
+6.  JavaScript in the tab application can parse the token and extract the information it needs, such as the user's email address. The tab app can optionally exchange the token server-side for further Graph permissions.
+
+### Problem Definition
+Teams SSO (https://docs.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/auth-aad-sso?tabs=dotnet) offers a seamless experience for partners to build auth inside of Teams. However, Teams SSO is only available for partners who use AzureAD already in their app. For partners that don't use AAD, they often have to use a workaround. There are two main types of scenarios that we typically see partners face when doing these workarounds:
+1.  the user exists in the third-party app with the same email address
+2.  the user exists in the third-party app with a different email address
+
+#### Scenario #1
+If the email addresses are same, then you can be linked right after the user is signed in with AAD. This is relatively simple for you to implement as you merely must search their own auth system for the user’s email and associated the access token with that email. You need to create a separate attribute or separate table to match a user’s email address with the AAD access token (and token refresh). 
+
+>To add to this difficulty, Microsoft also does not advise email mapping. Emails and User Principal Names (UPNs) can sometimes change.
+ 
+#### Scenario #2
+If the email addresses are different, you must do the following:
+1.  Ask the user to consent/sign with SSO
+2.  Ask the user to sign in with the email they use with your 3P app
+3.  Associate their Teams identity with their app’s identity in a table (or as an attribute on their database)
+Partners have been having some trouble with both of these scenarios. Scenario #2 is especially problematic for several partners. The underlying difficulty is associating a user’s AAD email and associated access token with a different email. For example, the user’s AAD email could be: john@contoso.com, but the email that is registered with the partner is john@gmail.com or even johnmatthews@contoso.com. In both of these examples, the emails don’t match up. The personal email address issue occurs frequently with apps that exist outside of enterprise scope (e.g., Starbucks, Uber, Coda, Observable). This is especially difficult to process as emails/UPNs can change as well for the same individual. 
```

Comment:
formatting is off here. I assume this should be a new paragraph but instead it runs on from 3. above